src/main/scala/model/actors/TroopActor.scala

```diff
@@ -0,0 +1,46 @@
+package model.actors
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import controller.GameLoopActor.GameLoopCommands.{EntitySpawned, EntityUpdated}
+import model.entities.*
+import model.actors.BulletActor
+import concurrent.duration.DurationInt
+
+object TroopActor:
+  def apply(troop: Troop): Behavior[ModelMessage] =
+    standardBehaviour(troop)
+
+  def standardBehaviour(troop: Troop): Behavior[ModelMessage] =
+    Behaviors.withTimers(timer => {
+      Behaviors.receive((ctx, msg) => {
+        msg match
+          case Update(elapsedTime, entities, replyTo) =>
+            val entityUpdated: Entity = troop.update(elapsedTime, entities)
```

Comment:
The problem is that the update is a method of the entity trait (as it should be, because is used by non-troop entities like bullets), thus we can not assume that the returned entity will be a Troop.